File: Python/Webcam Service/webcam_client.py
import numpy
import RobotRaconteur as RR
# Convenient shorthand
RRN = RR.RobotRaconteurNode.s
import cv2.cv as cv

def main():
    # Use NumPy
    RRN.UseNumPy = True
    
    # Register a Local Transport
    t1 = RR.LocalTransport()
    RRN.RegisterTransport(t1)
    
    # Register a TCP Transport
    t2 = RR.TcpTransport()
    RRN.RegisterTransport(t2)
    
    webcam = RRN.ConnectService('tcp://localhost:2345/example.webcam/Webcam');
    if webcam.camera_found > 0:
        cv.NamedWindow("Image")
        
        img_header = webcam.getImageHeader()
        c = img_header.channels
        if c == 1:
            current_image = cv.CreateMat(img_header.height, img_header.width, cv.CV_8UC1)
        elif c == 3:
            current_image = cv.CreateMat(img_header.height, img_header.width, cv.CV_8UC3)
        elif c == 4:
            current_image = cv.CreateMat(img_header.height, img_header.width, cv.CV_8UC4)
        
        webcam.StartStreaming()
        while True:
            RRimg = webcam.getCurrentImage()
            cv.SetData(current_image, RRimg.data.tostring())
            if (not current_image is None):
                cv.ShowImage("Image", current_image)
            if cv.WaitKey(50) > 0:
                break
        
        cv.DestroyAllWindows()
        webcam.StopStreaming()

    # THIS MUST BE HERE OR RISK SEGFAULT
    RRN.Shutdown()

if __name__ == '__main__':
    main()

File: MATLAB/webcam_client.m
clear variables; close all;

w = RobotRaconteur.Connect('tcp://localhost:2345/example.webcam/Webcam');

%%
im_h = w.getImageHeader();

figure(1); clf;
h = imshow(zeros(im_h.height,im_h.width,im_h.channels,'uint8'));

while (1)
    im = w.getCurrentImage();
    Ib = reshape(im.data(1:im_h.channels:end),[im.width im.height])';
    Ig = reshape(im.data(2:im_h.channels:end),[im.width im.height])';
    Ir = reshape(im.data(3:im_h.channels:end),[im.width im.height])';
    I = cat(3,Ir,Ig,Ib);
    figure(1);
    set(h,'CData',I);   
    pause(1/24);
    drawnow;
end
